packages/flutter_tools/test/commands.shard/hermetic/run_test.dart

```diff
@@ -542,6 +547,30 @@ void main() {
       expect(dartDefines, <String>['FLUTTER_WEB_AUTO_DETECT=false','FLUTTER_WEB_USE_SKIA=false']);
     });
   });
+
+  testUsingContext('Flutter run catches service has disappear errors and throws a tool exit', () async {
+    final FakeResidentRunner residentRunner = FakeResidentRunner();
```

Comment:
How awful would it be to get this to boil down to a fake vm service host that throws the rpc error?

Reply:
The resident runner does a tremendous amount of setup. It would be bad